webapp/src/dogma/features/repo/permissions/UserPermission.tsx

```diff
@@ -0,0 +1,77 @@
+import { Tag, TagLabel, Wrap, WrapItem } from '@chakra-ui/react';
+import {
+  MutationDefinition,
+  BaseQueryFn,
+  FetchArgs,
+  FetchBaseQueryError,
+  FetchBaseQueryMeta,
+} from '@reduxjs/toolkit/dist/query';
+import { MutationTrigger } from '@reduxjs/toolkit/dist/query/react/buildHooks';
+import { createColumnHelper } from '@tanstack/react-table';
+import { DynamicDataTable } from 'dogma/common/components/table/DynamicDataTable';
+import { DeleteMember } from 'dogma/features/metadata/DeleteMember';
+import { DeleteUserPermissionDto } from 'dogma/features/repo/permissions/DeleteUserPermissionDto';
+import { PerUserPermissionDto } from 'dogma/features/repo/RepoPermissionDto';
+import { useMemo } from 'react';
+
+export const UserPermission = ({
+  projectName,
+  repoName,
+  perUserPermissions,
+  deleteMember,
+  isLoading,
+}: {
+  projectName: string;
+  repoName: string;
+  perUserPermissions: PerUserPermissionDto;
+  deleteMember: MutationTrigger<
+    MutationDefinition<
+      DeleteUserPermissionDto,
+      BaseQueryFn<string | FetchArgs, unknown, FetchBaseQueryError, Record<string, never>, FetchBaseQueryMeta>,
+      'Metadata',
+      void,
+      'api'
+    >
+  >;
+  isLoading: boolean;
+}) => {
+  const columnHelper = createColumnHelper<[string, string[]]>();
+  const columns = useMemo(
+    () => [
+      columnHelper.accessor((row) => row[0], {
```

Comment:
I found it challenging to understand the nested square brackets. 
So I studied Typescript Type manipulation to deal with it. 😄
https://www.typescriptlang.org/docs/handbook/2/keyof-types.html
It is possible to extract the key and value type from `PerUserPermissionDto`
```ts
type K = keyof PerUserPermissionDto;
type UserAndPermission = [K, PerUserPermissionDto[K]];
// It complied.
const userAndPermission: UserAndPermission = ['hello', ['READ']];
// Failed to compile
const userAndPermission: UserAndPermission = ['hello', ['RED']];
```

If we use the derived type, the unnamed type could be replaced with it.
```suggestion
  const columnHelper = createColumnHelper<UserAndPermission>();
  const columns = useMemo(
    () => [
      columnHelper.accessor((row: UserAndPermission) => row[0], {
```